### PR TITLE
Fix overflow error for `*Vec` corpusfile-based training

### LINKED ISSUE #2258: Integer overflow during `FastText` training with `corpus_file`
#### Description

`model = FastText(corpus_file="sentences_norm.txt.gz", workers=14, iter=5, size=200, sg=1, hs=1)`

with the following sizes 

```
2018-11-05 16:57:52,809 : INFO : collected 6532860 word types from a corpus of 4728738902 raw words and 238627116 sentences
2018-11-05 16:57:52,809 : INFO : Loading a fresh vocabulary
2018-11-05 16:58:00,788 : INFO : effective_min_count=5 retains 1887156 unique words (28% of original 6532860, drops 4645704)
2018-11-05 16:58:00,788 : INFO : effective_min_count=5 leaves 4721157112 word corpus (99% of original 4728738902, drops 7581790)
2018-11-05 16:58:07,437 : INFO : deleting the raw counts dictionary of 6532860 items
2018-11-05 16:58:07,615 : INFO : sample=0.001 downsamples 26 most-common words
2018-11-05 16:58:07,615 : INFO : downsampling leaves estimated 3749158657 word corpus (79.4% of prior 4721157112)
2018-11-05 16:58:11,281 : INFO : constructing a huffman tree from 1887156 words
2018-11-05 16:59:36,077 : INFO : built huffman tree with maximum node depth 30
2018-11-05 17:00:17,300 : INFO : estimated required memory for 1887156 words, 1929637 buckets and 200 dimensions: 7871448352 bytes
2018-11-05 17:00:17,398 : INFO : resetting layer weights
2018-11-05 17:01:43,333 : INFO : Total number of ngrams is 1929637
2018-11-05 17:02:11,990 : INFO : training model with 14 workers on 1887156 vocabulary and 200 features, using sg=1 hs=1 sample=0.001 negative=5 window=5
```

yields 

```
Exception in thread Thread-2120:
Traceback (most recent call last):
  File "/home/joelkuiper/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/joelkuiper/anaconda3/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/joelkuiper/anaconda3/lib/python3.6/site-packages/gensim/models/base_any2vec.py", line 175, in _worker_loop_corpusfile
    total_examples=total_examples, total_words=total_words, **kwargs)
  File "/home/joelkuiper/anaconda3/lib/python3.6/site-packages/gensim/models/fasttext.py", line 561, in _do_train_epoch
    total_examples, total_words, work, neu1)
  File "gensim/models/fasttext_corpusfile.pyx", line 126, in gensim.models.fasttext_corpusfile.train_epoch_sg
OverflowError: value too large to convert to int
```` 

on all workers. Note that the sg and hs parameters seem to have no relation to this, also happens without them. 

### Steps to reproduce
`model = FastText(corpus_file="sentences_norm.txt.gz", workers=14, iter=5,size=200)`

#### Expected Results
Should train the model

#### Actual Results
Exception thrown, no further output. 

```
Traceback (most recent call last):
  File "/home/joelkuiper/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/joelkuiper/anaconda3/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/joelkuiper/anaconda3/lib/python3.6/site-packages/gensim/models/base_any2vec.py", line 175, in _worker_loop_corpusfile
    total_examples=total_examples, total_words=total_words, **kwargs)
  File "/home/joelkuiper/anaconda3/lib/python3.6/site-packages/gensim/models/fasttext.py", line 561, in _do_train_epoch
    total_examples, total_words, work, neu1)
  File "gensim/models/fasttext_corpusfile.pyx", line 126, in gensim.models.fasttext_corpusfile.train_epoch_sg
OverflowError: value too large to convert to int
```

#### Versions

Python 3.6.6 |Anaconda, Inc.| (default, Oct  9 2018, 12:34:16) 
[GCC 7.3.0]
NumPy 1.15.3
SciPy 1.1.0
gensim 3.6.0

On Ubuntu 16.04

**edit** seems to work fine when passing in a `LineSentence` object

### PR DESCRIPTION
**Pull request by bm371613**
Fix #2258 

Corpus size is currently limited in terms of the total word count to what fits a 32-bit signed integer. This can result in a `OverflowError: value too large to convert to int` for a big corpus. This PR replaces `int` with a `long long` to allow for bigger corpora.

### CODE REVIEW

**Comment by piskvorky** [comment]
Oh wow, how did that slip through. Thanks a lot for spotting & reporting!

File: gensim/models/doc2vec_corpusfile.pyx [hunk 1]
```diff
@@ -153,7 +153,8 @@ def d2v_train_epoch_dbow(model, corpus_file, offset, start_doctag, _cython_vocab
 
     cdef int i, j, document_len
     cdef int effective_words = 0
-    cdef int total_effective_words = 0, total_documents = 0, total_words = 0
+    cdef int total_effective_words = 0, total_documents = 0
```
Comment (piskvorky) [inline gensim/models/doc2vec_corpusfile.pyx]: How about changing these too, to be on the safe side (especially `total_effective_words`? Any reason not to?

**Comment by bm371613** [comment]
@piskvorky If "as expected" means accepting bigger values, a test could pass a big number and asserting there is no `OverflowError`. Is that what you meant?

File: gensim/models/doc2vec_corpusfile.pyx (hunk 1 above)
Comment (bm371613) [reply gensim/models/doc2vec_corpusfile.pyx]: There are `long long`s in many other places. Do you want explicit sizes everywhere?

Comment (piskvorky) [reply gensim/models/doc2vec_corpusfile.pyx]: I think it makes sense, but is not critical.

More important to me is that any variable that holds a potentially  large count (token frequency, number of tokens in a corpus, number of sentences in a corpus, …) is at least 64 bit.

**Comment by bm371613** [comment]
Note that CI failed due to flake8 errors on files not modified in this PR. A new version of flak8 was released a few days ago and it's likely that the CI would also fail on the target branch.

**Comment by menshikh-iv** [comment]
Big thanks @bm371613, can you please allow me to push into your branch (https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/#enabling-repository-maintainer-permissions-on-existing-pull-requests  (4)), I need that for fix CI

**Comment by menshikh-iv** [comment]
@bm371613 can you give me permissions manually to commit into your fork repo?

**Comment by bm371613** [comment]
@menshikh-iv ok done

**Comment by menshikh-iv** [comment]
@bm371613 it works, thank you!

**Comment by menshikh-iv** [comment]
@bm371613 awesome, thanks for the fix, congratz with the first contribution :1st_place_medal:

**Pull request merged**